test/hotspot/jtreg/runtime/valhalla/inlinetypes/field_layout/TestLayoutFlags.java

```diff
@@ -0,0 +1,276 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+ /*
+ * @test id=TestLayoutFlags_0
+ * @library /test/lib
+ * @modules java.base/jdk.internal.vm.annotation
+ * @enablePreview
+ * @compile FieldLayoutAnalyzer.java TestLayoutFlags.java
+ * @run main/othervm TestLayoutFlags 0
+ */
+
+ /*
+ * @test id=TestLayoutFlags_1
+ * @library /test/lib
+ * @modules java.base/jdk.internal.vm.annotation
+ * @enablePreview
+ * @compile FieldLayoutAnalyzer.java TestLayoutFlags.java
+ * @run main/othervm TestLayoutFlags 1
+ */
+
+ /* @test id=TestLayoutFlags_2
+ * @library /test/lib
+ * @modules java.base/jdk.internal.vm.annotation
+ * @enablePreview
+ * @compile FieldLayoutAnalyzer.java TestLayoutFlags.java
+ * @run main/othervm TestLayoutFlags 2
+ */
+
+ /* @test id=TestLayoutFlags_3
+ * @library /test/lib
+ * @modules java.base/jdk.internal.vm.annotation
+ * @enablePreview
+ * @compile FieldLayoutAnalyzer.java TestLayoutFlags.java
+ * @run main/othervm TestLayoutFlags 3
+ */
+
+/* @test id=TestLayoutFlags_4
+ * @library /test/lib
+ * @modules java.base/jdk.internal.vm.annotation
+ * @enablePreview
+ * @compile FieldLayoutAnalyzer.java TestLayoutFlags.java
+ * @run main/othervm TestLayoutFlags 4
+ */
+
+/* @test id=TestLayoutFlags_5
+ * @library /test/lib
+ * @modules java.base/jdk.internal.vm.annotation
+ * @enablePreview
+ * @compile FieldLayoutAnalyzer.java TestLayoutFlags.java
+ * @run main/othervm TestLayoutFlags 5
+ */
+
+/* @test id=TestLayoutFlags_6
+ * @library /test/lib
+ * @modules java.base/jdk.internal.vm.annotation
+ * @enablePreview
+ * @compile FieldLayoutAnalyzer.java TestLayoutFlags.java
+ * @run main/othervm TestLayoutFlags 6
+ */
+
+/* @test id=TestLayoutFlags_7
+ * @library /test/lib
+ * @modules java.base/jdk.internal.vm.annotation
+ * @enablePreview
+ * @compile FieldLayoutAnalyzer.java TestLayoutFlags.java
+ * @run main/othervm TestLayoutFlags 7
+ */
+
+ import java.util.ArrayList;
+ import java.util.Collections;
+ import java.util.List;
+
+ import jdk.internal.vm.annotation.ImplicitlyConstructible;
+ import jdk.internal.vm.annotation.LooselyConsistentValue;
+ import jdk.internal.vm.annotation.NullRestricted;
+
+ import java.lang.reflect.Method;
+ import java.lang.reflect.Modifier;
+ import jdk.test.lib.Asserts;
+ import jdk.test.lib.process.OutputAnalyzer;
+ import jdk.test.lib.process.ProcessTools;
+
+ public class TestLayoutFlags {
+
+   static class TestRunner {
```

Comment:
Just a quick drive-by comment: Java code should use 4-whitespace indentation. Same for other files.